src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,55 +1,130 @@
-import React from 'react';
+import React, {
+  FC,
+  useState,
+  ChangeEvent,
+  FormEvent,
+} from 'react';
 import './FindMovie.scss';
-
+import * as cx from 'classnames';
 import { MovieCard } from '../MovieCard';
-import movies from '../../api/movies.json';
+import { getData, API_URL } from '../../utils/api';
+import { Movie } from '../../utils/interfaces';
+
+
+interface Props {
+  addMovie: (movie: Movie) => void;
+}
+
+export const FindMovie: FC<Props> = ({ addMovie }) => {
+  const [inputValue, setInputValue] = useState('');
+  const [previewMovie, setPreviewMovie] = useState<Movie | null>(null);
+  const [isError, setError] = useState(false);
+
+  const searchMovie = async () => {
+    const getNewMovie = () => {
+      return getData(`${API_URL}${inputValue}`);
+    };
+
+    const newMovieFromServer = await getNewMovie();
+
+    const {
+      Title: title,
+      Plot: description,
+      Poster: imgUrl,
+      imdbID: imdbId,
+      Response,
+    } = newMovieFromServer;
+
+    if (Response === 'True') {
+      const imdbUrl = API_URL + imdbId;
+      const newMovie = {
+        title,
+        description,
+        imgUrl,
+        imdbId,
+        imdbUrl,
+      };
+
+      setPreviewMovie(newMovie);
+      setError(false);
+    } else {
+      setError(true);
+    }
+  };
 
-export const FindMovie = () => (
-  <>
-    <form className="find-movie">
-      <div className="field">
-        <label className="label" htmlFor="movie-title">
+  const inputHandler = (event: ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setInputValue(value.replace(/^\s/, ''));
+  };
+
+  const submitHandler = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (previewMovie) {
+      addMovie(previewMovie);
+      setInputValue('');
+      setError(false);
+      setPreviewMovie(null);
+    }
+  };
+
+  return (
+    <>
+      <form className="find-movie" onSubmit={submitHandler}>
+        <div className="field">
+          <label className="label" htmlFor="movie-title">
           Movie title
-        </label>
-
-        <div className="control">
-          <input
-            type="text"
-            id="movie-title"
-            placeholder="Enter a title to search"
-            className="input is-danger"
-          />
-        </div>
+          </label>
 
-        <p className="help is-danger">
-          Can&apos;t find a movie with such a title
-        </p>
-      </div>
+          <div className="control">
+            <input
+              type="text"
+              id="movie-title"
+              placeholder="Enter a title to search"
+              className={cx.default({
+                input: true,
```

Comment:
if u use "Error(bool)" in state, maybe it will be better to "setError" in ChangeHandler, and check 'error' here, instead of `...input: true...`